mods/default/nodes.lua

```diff
@@ -586,30 +586,121 @@ minetest.register_node("default:snow", {
 	inventory_image = "default_snowball.png",
 	wield_image = "default_snowball.png",
 	paramtype = "light",
+	paramtype2 = "leveled",
 	buildable_to = true,
 	floodable = true,
+	walkable = true,
 	drawtype = "nodebox",
+	leveled = 7,
 	node_box = {
-		type = "fixed",
-		fixed = {
-			{-0.5, -0.5, -0.5, 0.5, -0.25, 0.5},
-		},
+		type = "leveled",
+		fixed = {-0.5, -0.5, -0.5, 0.5, -0.5, 0.5},
 	},
 	collision_box = {
-		type = "fixed",
-		fixed = {
-			{-0.5, -0.5, -0.5, 0.5, -7 / 16, 0.5},
-		},
+		type = "leveled",
+		fixed = {-0.5, -0.5, -0.5, 0.5, -0.5, 0.5},
+	},
+	selection_box = {
+		type = "leveled",
+		fixed = {-0.5, -0.5, -0.5, 0.5, -0.5, 0.5},
 	},
 	groups = {crumbly = 3, falling_node = 1, snowy = 1},
 	sounds = default.node_sound_snow_defaults(),
 
 	on_construct = function(pos)
-		pos.y = pos.y - 1
-		if minetest.get_node(pos).name == "default:dirt_with_grass" then
-			minetest.set_node(pos, {name = "default:dirt_with_snow"})
+		pos.y = pos.y-1
+		local node = minetest.get_node(pos)
+		if node.name == "default:dirt"
+		or node.name == "default:dirt_with_grass"
+        or node.name == "default:dirt_with_grass_footsteps"
+        or node.name == "default:dirt_with_rainforest_litter"
+        or node.name == "default:dirt_with_coniferous_litter" then
```

Comment:
Please read https://dev.minetest.net/Lua_code_style_guidelines.

- Indentation is done with tabs.
- if foo or
\<two tabs\>bar then
\<one tab\>-- do something
end

Reply:
Darned text editor xD
I've fixed it.